### DISCUSSION TITLE
Keep getting "ERROR: Invalid token provided!" How exactly do I get a VALID token?

### DISCUSSION THREAD

**Discussion by PacoH** [vzhd1701/enex2notion#81] [Q&A] [answered]
I learned about this Evernote importer on [Leaving Evernote | John’s web site](https://www.johnborwick.com/2022/07/31/leaving-evernote.html).

I tried to convert an `enex` file to Notion format:

	enex2notion --token "$TOKEN" --add-meta --mode PAGE --condense-lines --done-file done-tracking "$f"
	ERROR: Invalid token provided!

I tried several methods to get a token and each time I got a different invalid token:

1. [How To Find Your Notion v2 Token — Red Gregory](https://www.redgregory.com/notion/2020/6/15/9zuzav95gwzwewdu1dspweqbv481s5)

Use Google Chrome, and sign into your Notion workspace.
Right click anywhere inside the page and select "Inspect."
Locate "Application" and select "Cookies." Here you should be able to find "token_v2." Copy the property next to it called "Value." I got the error using this value.

2. [Authorization](https://developers.notion.com/docs/authorization)

It involves creating an `integration`, which creates an `Internal Integration Token` that starts with `secret_`, which can be copied, and then sharing a page or database with your integration using the API. This token did not work either but I did not share a page or database with my internal integration first because I don't have any pages yet and don't feel the need to learn the API enough to create said page, just to get a token so I can covert Evernote files to Notion format. I've already spent hours on this and I can't help thinking that this should not be this difficult.

So what is the simplest way to get a *valid* token so I can convert `enex` files to Notion format for import?

**Comment by vzhd1701** [comment, marked as answer]
Make sure that you copy your token completely, it should be 162 characters long. See this article on how to get the token in any browser
https://vzhd1701.notion.site/Find-Your-Notion-Token-5f57951434c1414d84ac72f88226eede
